colossalai/pipeline/schedule/zero_bubble_pp.py

```diff
@@ -472,19 +478,47 @@ def backward_b_step(
         # calculate bwd b step ; only dx = w*dy;
 
         # Retain the grad on the input_obj.
-        tree_map(retain_grad, input_obj)
+        if input_obj is None:
+            return None
+        else:
+            tree_map(retain_grad, input_obj)
+
+        # x, y, dy list for backward_by_grad; Type: list[tensor];
+        input_obj_ = []
+        output_obj_ = []
+        output_obj_grad_ = []
+
+        # get x from input_obj to input_obj_
+        for k, v in input_obj.items():
```

Comment:
should use tree_map or tree_flatten

Reply:
> should use tree_map or tree_flatten

Fixed in https://github.com/hpcaitech/ColossalAI/pull/6065/commits/c6d6ee39bda6d0e3aa0a6233796a0d1059eb30dc;